src/containers/Petitions.js

```diff
@@ -7,8 +7,14 @@ import Petitions from 'components/Petitions';
 import getPetitions from 'selectors/petitions';
 
 const PetitionsContainer = React.createClass({
-  componentDidMount () {
-    this.props.fetchPetitions(this.props);
+  // When the component gets added to the DOM,
+  // fetch Petitions if none are defined (clientside)
+  componentWillMount () {
+    const { petitions } = this.props;
+
+    if (!petitions.length) {
+      this.props.fetchPetitions(this.props);
+    }
```

Comment:
This will break client side pagination. First page's petitions won't be replaced by those from second page. I guess we're good to always fetch petitions here.